infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/fsm/sink/LogReplicationSinkEvent.java

```diff
@@ -0,0 +1,38 @@
+package org.corfudb.infrastructure.logreplication.runtime.fsm.sink;
+
+import lombok.Getter;
+
+/**
+ * This class represents events for SINK to establish a replication channel with the SOURCE, when SINK is the connection
+ * initiator
+ */
+public class LogReplicationSinkEvent {
+
+    /* Enum listing the various type of LogReplicationSinkEvent.
+     */
+    public enum LogReplicationSinkEventType {
```

Comment:
These events need to have a lot more comments for these like when the event happens and what are the actions associated. Ideally we generate ascii art for the state machine.